### PR TITLE
Test with jest

### PR DESCRIPTION
**Pull request by jodom**
This PR closes first part of #62.

It reconsiders the tests and testing infrastructure, proposing [Jest](https://jestjs.io/) test runner and [React Testing Library](https://testing-library.com/docs/react-testing-library/intro).
Jest gives an advantage of being able to run non-react / non-ui tests so we can use the same runner to **test services** and other utils. React testing lib has an interesting philosophy of [testing user actions and interactivity](https://testing-library.com/docs/guiding-principles), rather than component state.
Tests introduced don't provide adequate coverage; Additions to the suites introduced here is most welcome :)
CI loop was not investigated.

### CODE REVIEW

File: src/components/Menu.test.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import { render, cleanup } from '@testing-library/react';
+import { Menu } from './Menu';
+
+describe('Menu tests', () => {
+    it('renders the color picker', () => {
```
Comment (stellasia) [inline src/components/Menu.test.js:6]: Copy / paste here ;)

File: src/components/Layer.js [hunk 2]
```diff
@@ -675,7 +685,7 @@ export class Layer extends Component {
 							{this.renderConfigCypher()}
 							{this.renderConfigSpatial()}
 
-							<h4> > Map rendering</h4>
+							<h4>{' > Map rendering'}</h4>
```
Comment (stellasia) [inline src/components/Layer.js:688]: Just out of curiosity, why this change (and the others of the same kind above)?

File: src/components/ColorPicker.js [hunk 3]
```diff
@@ -72,5 +72,3 @@ class ColorPicker extends React.Component {
         )
     }
 }
-
-export default ColorPicker;
```
Comment (stellasia) [inline src/components/ColorPicker.js:76]: Again, just for me to understand, any reason not to use the default export?

**Review by stellasia — APPROVED** [review]
I really like this new test suite! With the examples you implemented, I should be able to increase the coverage. Just mentioned a copy/paste error, but that can be fixed when tests get extended.

**Comment by stellasia** [comment]
FYI, I pushed a small update to the master branch to trigger Actions on PR. You'll have to update your branch (can wait until your next PR)

Thanks for this nice PR!

File: src/components/Menu.test.js (hunk 1 above)
Comment (jodom) [reply src/components/Menu.test.js:6]: good catch :)

File: src/components/Layer.js (hunk 2 above)
Comment (jodom) [reply src/components/Layer.js:688]: Cheap syntactical fix suggested by my editor. > in the string as written before errors it clashes with html tags closing >

Comment (stellasia) [reply src/components/Layer.js:688]: Ok, good reason :)

File: src/components/ColorPicker.js (hunk 3 above)
Comment (jodom) [reply src/components/ColorPicker.js:76]: Preferring named exports for module control. default exports are actually okay, but with multiple exports from same module, it forces the import syntax like `import Module, { OtherExports } from 'module'`. With named exports, we always do `import { Module, OtherExports } from 'module'`

Comment (stellasia) [reply src/components/ColorPicker.js:76]: Ok, thanks for the explanation!
